minari/utils.py

```diff
@@ -161,31 +163,27 @@ def validate_datasets_to_combine(datasets_to_combine: List[MinariDataset]) -> En
 
     Returns:
         combined_dataset_env_spec (EnvSpec): the resulting EnvSpec of combining the MinariDatasets
-    """
-    assert all(isinstance(dataset, MinariDataset) for dataset in datasets_to_combine)
-
-    # Check if there are any `None` max_episode_steps
-    if any(
-        (dataset.spec.env_spec.max_episode_steps is None)
-        for dataset in datasets_to_combine
-    ):
-        max_episode_steps = None
-    else:
-        max_episode_steps = max(
-            dataset.spec.env_spec.max_episode_steps for dataset in datasets_to_combine
-        )
 
-    combine_env_spec = []
+    """
+    common_env_spec = copy.deepcopy(datasets_to_combine[0].spec.env_spec)
     for dataset in datasets_to_combine:
-        dataset_env_spec = copy.deepcopy(dataset.spec.env_spec)
-        dataset_env_spec.max_episode_steps = max_episode_steps
-        combine_env_spec.append(dataset_env_spec)
-
-    assert all(
-        env_spec == combine_env_spec[0] for env_spec in combine_env_spec
-    ), "The datasets to be combined have different values for `env_spec` attribute."
+        assert isinstance(dataset, MinariDataset)
+        env_spec = dataset.spec.env_spec
+        if env_spec is not None:
+            assert (
+                common_env_spec is not None
+            ), "Found incompatible env_spec in datasets"
+            if (
+                common_env_spec.max_episode_steps is None
+                or env_spec.max_episode_steps is None
+            ):
+                common_env_spec.max_episode_steps = None
+            else:
+                common_env_spec.max_episode_steps = max(
+                    common_env_spec.max_episode_steps, env_spec.max_episode_steps
+                )
```

Comment:
this sounds good, but it still doesn't check if the two env_spec are the same.
One thing you can do is creating a copy of env_spec, modify the max_episode_steps to be the same of common_env_spec and then check env_spec == common_env_spec

Reply:
yes, that's correct. I'm doing the same in lines after 184. (the edit comment that I mentioned)

Reply:
Added the fix.